### ISSUE TITLE
Validation curve plot function

### ISSUE THREAD

**Issue by Hellengeremias** [topepo/caret#831]
Dear all,

I'm new on predictive modelling area and I'm studying applied problems on caret package.

I would like to know if it is possible visualize a "validation curve", similar to those avalilable on Python (Sklearn), that includes code creating validation curves by varying hyperparamaters and computing and plotting performance metrics (like auc or rmse) during training process. 

It's not very clear to me how I can save some metrics during training process (using function "train") on caret, additionaly to those results related to cross validation (for example).

I would like to plot these metrics to see how the trainings process evolved and
how far cross validation results is from the training ones across hyperparameters values. Is it possible?
In your opinion, this kind of plot could be useful to analyse overfitting on training process? 

Best regards,
Hellen

**Comment by topepo** [comment]
It sounds like what the `ggplot` (and `plot`) functions in `caret` do for `train` objects.  See [section 5.5.3 of the docs](https://topepo.github.io/caret/model-training-and-tuning.html#basic-parameter-tuning) as an example. 

Does that match what you are looking for?

**Comment by Hellengeremias** [comment]
Dear @topepo,

Thank you for your quickly reply. These functions shows resamples results.
It is exatly what I want, but additionaly to resampling profile, I would like to visualize training results.

For example, if I choose cross validation method, I would like to loop over 'i' tuning parameters and 'k' folds, create a predictive model for this, get the estimates of performance for training and test separately, compute and plot the average of these estimates through 'k' folds for train and test (y-axis) vs 'i' tuning parametes (x-axis).

Here is a python output available on https://gigadom.wordpress.com/2017/10/29/practical-machine-learning-with-r-and-python-part-4/.

![svm validation curve](https://user-images.githubusercontent.com/22616401/36164247-f7ac250a-10d2-11e8-89c6-80ef0e9e15f1.jpg)

Thank you once again!

Bes regards,
Hellen

**Comment by topepo** [comment]
I haven't implemented this because, frankly, it isn't a great idea. We know that, for the majority of models, the statistics that you get by simply re-predicting the training set can be horribly optimistic (see the "apparent error rate"). Linear models may be an exception but otherwise I don't think that this is worth adding.

**Comment by Hellengeremias** [comment]
Thank you @topepo for your reply!
